auction-server/src/opportunity/service/get_quote.rs

```diff
@@ -50,25 +44,82 @@ pub struct GetQuoteInput {
     pub quote_create: entities::QuoteCreate,
 }
 
+pub fn get_quote_permission_key(
+    tokens: &entities::QuoteTokens,
+    user_wallet_address: &Pubkey,
+) -> Pubkey {
+    // get pda seeded by user_wallet_address, mints, and token amount
+    let input_token_amount: [u8; 8];
+    let output_token_amount: [u8; 8];
+    let seeds = match tokens {
+        entities::QuoteTokens::InputTokenSpecified {
+            input_token,
+            output_token,
+        } => {
+            let input_token_mint = input_token.token.as_ref();
+            let output_token_mint = output_token.as_ref();
+            input_token_amount = input_token.amount.to_le_bytes();
+            [
+                user_wallet_address.as_ref(),
+                input_token_mint,
+                input_token_amount.as_ref(),
+                output_token_mint,
+            ]
+        }
+        entities::QuoteTokens::OutputTokenSpecified {
+            input_token,
+            output_token,
+        } => {
+            let input_token_mint = input_token.as_ref();
+            let output_token_mint = output_token.token.as_ref();
+            output_token_amount = output_token.amount.to_le_bytes();
+            [
+                user_wallet_address.as_ref(),
+                input_token_mint,
```

Comment:
this seems weird. I think I'd rather have a 1 bytes discriminator between InputTokenSpecified and OutputTokenSpecified

Reply:
yeah that's fair, i guess the main difference here is the placement of the `u64`. i can add a string discriminator at the start to be explicit like "input_token_specified" or "output_token_specified", although these should be distinct already